src/blockdata/opcodes.rs

```diff
@@ -652,48 +652,99 @@ impl fmt::Debug for All {
             all::OP_CLTV => write!(f, "CLTV"),
             all::OP_CSV => write!(f, "CSV"),
             All {code: x} if x >= all::OP_NOP1.code && x <= all::OP_NOP10.code => write!(f, "NOP{}", x - all::OP_NOP1.code + 1),
+            all::OP_INVALIDOPCODE => write!(f, "INVALIDOPCODE"),
+            all::OP_CHECKSIGADD => write!(f, "CHECKSIGADD"),
             All {code: x} => write!(f, "RETURN_{}", x),
         }
     }
 }
 
+
+/// Classification context for the opcode. Some opcodes like `OP_RESERVED`
+/// abort the script in [`ClassifyContext::Legacy`] context, but will act as OP_SUCCESS in tapscript
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ClassifyContext {
+    /// Opcode used in tapscript context
+    TapScript,
+    /// Opcode used in legacy context
+    Legacy,
+}
+
 impl All {
     /// Classifies an Opcode into a broad class
     #[inline]
-    pub fn classify(self) -> Class {
-        // 17 opcodes
-        if self == all::OP_VERIF || self == all::OP_VERNOTIF ||
-           self == all::OP_CAT || self == all::OP_SUBSTR ||
-           self == all::OP_LEFT || self == all::OP_RIGHT ||
-           self == all::OP_INVERT || self == all::OP_AND ||
-           self == all::OP_OR || self == all::OP_XOR ||
-           self == all::OP_2MUL || self == all::OP_2DIV ||
-           self == all::OP_MUL || self == all::OP_DIV || self == all::OP_MOD ||
-           self == all::OP_LSHIFT || self == all::OP_RSHIFT {
-            Class::IllegalOp
-        // 11 opcodes
-        } else if self == all::OP_NOP ||
-                  (all::OP_NOP1.code <= self.code &&
-                   self.code <= all::OP_NOP10.code) {
-            Class::NoOp
-        // 75 opcodes
-        } else if self == all::OP_RESERVED || self == all::OP_VER || self == all::OP_RETURN ||
-                  self == all::OP_RESERVED1 || self == all::OP_RESERVED2 ||
-                  self.code >= all::OP_RETURN_186.code {
-            Class::ReturnOp
-        // 1 opcode
-        } else if self == all::OP_PUSHNUM_NEG1 {
-            Class::PushNum(-1)
-        // 16 opcodes
-        } else if all::OP_PUSHNUM_1.code <= self.code &&
-                  self.code <= all::OP_PUSHNUM_16.code {
-            Class::PushNum(1 + self.code as i32 - all::OP_PUSHNUM_1.code as i32)
-        // 76 opcodes
-        } else if self.code <= all::OP_PUSHBYTES_75.code {
-            Class::PushBytes(self.code as u32)
-        // 60 opcodes
-        } else {
-            Class::Ordinary(Ordinary::try_from_all(self).unwrap())
+    pub fn classify(self, ctx: ClassifyContext) -> Class {
+        // 3 opcodes
```

Comment:
Nit: we can do `use all::*` here and get rid of a lot of visual clutter with `all::` prefixes